File: examples/animation-loop/noiseloop2d.js
let duration = 3;
let scale = 0.3;

function setup(c) {}

function draw(c) {
  c.clear();
  let theta = c.loopAngle(duration);

  for (let x = 0; x < c.width; x++) {
    for (let y = 0; y < c.height; y++) {
      let noise = c.noiseLoop2D(theta, 1, x * scale, y * scale);
      let color = c.map(noise, 0, 1, 232, 255);
      c.stroke("§", color, 0);
      c.point(x, y);
    }
  }
}

function onKey(c, e) {
  if (e.Key == " ") {
    c.noiseSeed(Date.now());
  }
}

function onMouse(c, e) {
  // Handle mouse events here
}

File: examples/noiseloop2d/noiseloop2d.go
package main

import (
	"context"
	"strconv"
	"time"

	"github.com/emprcl/runal"
)

const (
	duration = 3
	scale    = 0.3
)

func main() {
	runal.Run(context.Background(), setup, draw, runal.WithOnKey(onKey))
}

func setup(c *runal.Canvas) {}

func draw(c *runal.Canvas) {
	c.Clear()
	theta := c.LoopAngle(duration)

	for x := 0; x < c.Width; x++ {
		for y := 0; y < c.Height; y++ {
			noise := c.NoiseLoop2D(
				theta,
				1,
				int(float64(x)*scale),
				int(float64(y)*scale),
			)
			color := c.Map(noise, 0, 1, 232, 255)
			colorStr := strconv.FormatFloat(color, 'f', -1, 64)
			c.Stroke("§", colorStr, "0")
			c.Point(x, y)
		}
	}
}

func onKey(c *runal.Canvas, e runal.KeyEvent) {
	if e.Key == "space" {
		c.NoiseSeed(time.Now().Unix())
	}
}
